File: Shared/Students.Models/ReferenceModels/KindOrder.cs
namespace Students.Models.ReferenceModels;

/// <summary>
/// Вид приказа
/// По хорошему это можно вынести в базовый дженерик класс Справочник и наследовать от него не плодя свойства одни и те же (кому-то предать на доработку)
/// </summary>
public class KindOrder
{
  /// <summary>
  /// Id программы
  /// </summary>
  public Guid Id { get; set; }

  /// <summary>
  /// Имя программы
  /// </summary>
  public string? Name { get; set; }
}

File: Client/src/storage/catalogConfigs/kindOrder.js
import {
    useGetAllAsync,
    useGetAllPagedAsync,
    useGetOneByIdAsync,
    useAddOneAsync,
    useEditOneAsync,
    useRemoveOneAsync,
} from '../crud/kindOrderCrud.js';
import { kindOrderModel } from '../models/index.js';

export default {
    detailsLink: 'kindOrder',
    hasDetailsPage: false,
    serverPaged: false,
    properties: kindOrderModel,
    crud: {
        useGetAllAsync,
        useGetAllPagedAsync,
        useGetOneByIdAsync,
        useAddOneAsync,
        useEditOneAsync,
        useRemoveOneAsync,
    },
    columns: [
        {
            title: 'Тип приказа',
            dataIndex: 'name',
            key: 'name',
        },
    ],
    dataConverter: (data) => data,
};